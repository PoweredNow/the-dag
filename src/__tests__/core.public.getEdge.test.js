const addEdges = require('../core/public/addEdges');
const addNodes = require('../core/public/addNodes');
const getEdge = require('../core/public/getEdge');

const getStateManipulators = require('../core/private/_getStateManipulators');
const { getState } = require('../core/private/_state');

describe('core.public.getEdge', () => {
  test('exports', () => {
    expect(getEdge).toMatchSnapshot();
  });
  test('gets edge by source and target objects', () => {
    const stateManipulators = getStateManipulators();
    addNodes([{ nodeID: 1 }, { nodeID: 2 }], stateManipulators);
    addEdges(
      [{ source: { nodeID: 1 }, target: { nodeID: 2 } }],
      stateManipulators
    );
    const edge = getEdge(
      { source: { nodeID: 1 }, target: { nodeID: 2 } },
      stateManipulators
    );
    expect(edge).toMatchSnapshot();
    const state = stateManipulators.getState();
    expect(state.edges).toMatchSnapshot();
  });
  test('gets edge by sourceID and targetID objects', () => {
    addNodes([{ nodeID: 1 }, { nodeID: 2 }]);
    addEdges([{ source: { nodeID: 1 }, target: { nodeID: 2 } }]);
    const edge = getEdge({ source: 1, target: 2 });
    expect(edge).toMatchSnapshot();
    const state = getState();
    expect(state.edges).toMatchSnapshot();
  });
});
