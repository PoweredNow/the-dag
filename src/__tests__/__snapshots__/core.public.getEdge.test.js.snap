// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`core.public.getEdge exports 1`] = `[Function]`;

exports[`core.public.getEdge gets edge by source and target objects 1`] = `
Object {
  "edgeData": Object {},
  "edgeID": "1_2",
  "sourceID": 1,
  "targetID": 2,
}
`;

exports[`core.public.getEdge gets edge by source and target objects 2`] = `
Object {
  "1_2": Object {
    "edgeData": Object {},
    "edgeID": "1_2",
    "sourceID": 1,
    "targetID": 2,
  },
}
`;

exports[`core.public.getEdge gets edge by sourceID and targetID objects 1`] = `
Object {
  "edgeData": Object {},
  "edgeID": "1_2",
  "sourceID": 1,
  "targetID": 2,
}
`;

exports[`core.public.getEdge gets edge by sourceID and targetID objects 2`] = `
Object {
  "1_2": Object {
    "edgeData": Object {},
    "edgeID": "1_2",
    "sourceID": 1,
    "targetID": 2,
  },
}
`;
